### DISCUSSION TITLE
Is there a way to import and use css variables?

### DISCUSSION THREAD

**Discussion by dollusion** [cypress-io/cypress#28080] [Questions and Help]
Basically I am checking if elements have the right color but I don't want my tests to break if I change colors in the css. For my app im using a global css file with variables like:
```
--dark-green: #205337;
--light-green: #95B46A;
--light-red: #DB504A;
```
and use them in code like: `var(--dark-green)`
Can I somehow import my css file to a cypress test and use it's variables?

**Comment by portyu9** [comment]
Hi @dollusion 

Yes. I would read the CSS custom property from the application at runtime rather than duplicate its value in the test.

One subtlety is that your variable may contain something like `#205337`, while the browser's computed `color` value is normally normalized to something like `rgb(...)`. You can let the browser resolve both:

```js
cy.document().then((doc) => {
  const probe = doc.createElement('span')

  probe.style.color = 'var(--dark-green)'
  doc.body.appendChild(probe)

  const expectedColor = getComputedStyle(probe).color
  probe.remove()

  cy.get('[data-cy="target"]')
    .should('have.css', 'color', expectedColor)
})
```

That way the test depends on the semantic variable `--dark-green`, not on its current hex value. Changing the variable's value in your global CSS will not require changing the test.

You can also read a custom property directly with:

```js
getComputedStyle(document.documentElement)
  .getPropertyValue('--dark-green')
  .trim()
```

Importing the CSS file into the Cypress spec is generally unnecessary here—the application has already loaded the CSS, and reading the runtime computed style tests what the browser is actually using.

Please mark this answer as accepted if it helped, thank you.
